File: examples/python/complete_flow.py
import os
import time
import requests

api_key = os.environ.get("EVOLINK_API_KEY")
if not api_key:
    raise SystemExit("Set EVOLINK_API_KEY first")

headers = {
    "Authorization": f"Bearer {api_key}",
    "Content-Type": "application/json",
    "X-EvoLink-Source": "skill",
    "X-EvoLink-Skill": "krea-2-turbo-image",
    "X-EvoLink-Package": "evolink-krea-2-turbo",
    "X-EvoLink-Campaign": "krea-2-turbo-image",
    "X-EvoLink-Touchpoint": "first-call",
}

create_resp = requests.post(
    "https://api.evolink.ai/v1/images/generations",
    headers=headers,
    json={
        "model": "krea-2-turbo",
        "prompt": "A cinematic product poster, silver headphones floating against a deep matte-black backdrop with soft rim lighting",
        "size": "16:9",
        "quality": "1K",
        "nsfw_check": False,
    },
    timeout=60,
)

if create_resp.status_code >= 400:
    raise SystemExit(f"Create task failed: {create_resp.status_code} {create_resp.text}")

task = create_resp.json()
task_id = task.get("id")
if not task_id:
    raise SystemExit(f"Create task did not return id: {task}")

for _ in range(120):
    poll_resp = requests.get(
        f"https://api.evolink.ai/v1/tasks/{task_id}",
        headers=headers,
        timeout=30,
    )
    if poll_resp.status_code >= 400:
        raise SystemExit(f"Poll failed: {poll_resp.status_code} {poll_resp.text}")

    task = poll_resp.json()
    status = task.get("status")
    if status == "completed":
        print(task)
        raise SystemExit(0)
    if status == "failed":
        raise SystemExit(f"Task failed: {task}")
    time.sleep(3)

raise SystemExit(f"Timed out waiting for task {task_id}")

File: examples/javascript/complete-flow.mjs
const apiKey = process.env.EVOLINK_API_KEY;
if (!apiKey) {
  throw new Error("Set EVOLINK_API_KEY first");
}

async function requestJson(url, options = {}) {
  const response = await fetch(url, options);
  const text = await response.text();
  const data = text ? JSON.parse(text) : {};
  if (!response.ok) {
    throw new Error(`${response.status} ${response.statusText}: ${JSON.stringify(data)}`);
  }
  return data;
}

const task = await requestJson("https://api.evolink.ai/v1/images/generations", {
  method: "POST",
  headers: {
    Authorization: `Bearer ${apiKey}`,
    "Content-Type": "application/json",
    "X-EvoLink-Source": "skill",
    "X-EvoLink-Skill": "krea-2-turbo-image",
    "X-EvoLink-Package": "evolink-krea-2-turbo",
    "X-EvoLink-Campaign": "krea-2-turbo-image",
    "X-EvoLink-Touchpoint": "first-call"
  },
  body: JSON.stringify({
  "model": "krea-2-turbo",
  "prompt": "A cinematic product poster, silver headphones floating against a deep matte-black backdrop with soft rim lighting",
  "size": "16:9",
  "quality": "1K",
  "nsfw_check": false
})
});

if (!task.id) {
  throw new Error(`Create task did not return id: ${JSON.stringify(task)}`);
}

for (let attempt = 0; attempt < 120; attempt += 1) {
  const current = await requestJson(`https://api.evolink.ai/v1/tasks/${task.id}`, {
    headers: {
      Authorization: `Bearer ${apiKey}`,
      "X-EvoLink-Source": "skill",
      "X-EvoLink-Skill": "krea-2-turbo-image",
      "X-EvoLink-Package": "evolink-krea-2-turbo",
      "X-EvoLink-Campaign": "krea-2-turbo-image",
      "X-EvoLink-Touchpoint": "first-call"
    }
  });

  if (current.status === "completed") {
    console.log(JSON.stringify(current, null, 2));
    process.exit(0);
  }
  if (current.status === "failed") {
    throw new Error(`Task failed: ${JSON.stringify(current)}`);
  }
  await new Promise((resolve) => setTimeout(resolve, 3000));
}

throw new Error(`Timed out waiting for task ${task.id}`);
